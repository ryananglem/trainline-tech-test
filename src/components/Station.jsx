import React from 'react'
import './station.css'

export const Station = ({scheduledAt, isTrainHere, station, 
                        isDestination, isOrigin, 
                        hasDeparted, estimatedAt}) => (
    <div className="stationContainer">
      <div className={(hasDeparted ? "" : "stationHighlight ") + "stationTime"}>
                {scheduledAt} <br/>
                {isTrainHere ? 
                    <div className="stationTrainHere">
                        <i className="material-icons md-18">train</i>
                    </div> : 
                    null}
       </div> 
       <div className={isDestination ? "stationLastCircle" : "stationCircle" + (isOrigin ? " stationFilled" : "")}>&nbsp;</div> 
       <div className="stationInfo">            
            <div className={hasDeparted ? "" : "stationHighlight"}>
               { station }
            </div>
            <div className="stationTimeInfo">
                { (estimatedAt!=="On time" ? 
                    (hasDeparted ? "Dept. " : "Exp. ") :
                    "") 
                   + estimatedAt }   
            </div>
        </div>        
    </div>
)
