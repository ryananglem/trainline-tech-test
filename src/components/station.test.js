import React from 'react'

import { Station } from './Station'
import renderer from 'react-test-renderer'
import Enzyme, {shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

describe("station", () =>{
    Enzyme.configure({ adapter: new Adapter() })
   
    const station = {
        "hasDeparted": true,
        "isTrainHere": true,
        "station": "Sevenoaks",
        "scheduledAt": "10:00",
        "estimatedAt": "On time",
        "isOrigin": true,
        "isDestination": false
      }
      const component = shallow(<Station
          hasDeparted={station.hasDeparted}
          isTrainHere={station.isTrainHere}
          scheduledAt={station.scheduledAt}
          estimatedAt={station.estimatedAt}
          onTime={station.onTime}
          isOrigin={station.isOrigin}
          isDestination={station.isDestination}                                                            
          station={station.station} 
                  />)
      
    it("renders correctly", () => {
        expect(component).toMatchSnapshot()
    })

    it("displays the train", () =>{
        expect(component.exists(<i className="material-icons md-18">train</i>)).toBe(true)
    })
    
    it("displays has departed time", () => {
        expect(component.exists("Dept. 10:00")).toBe(true)
    })
})