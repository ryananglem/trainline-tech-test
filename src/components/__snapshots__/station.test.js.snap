// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`station renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    estimatedAt="On time"
    hasDeparted={true}
    isDestination={false}
    isOrigin={true}
    isTrainHere={true}
    onTime={undefined}
    scheduledAt="10:00"
    station="Sevenoaks"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stationTime"
>
          10:00
           
          <br />
          <div
                    className="stationTrainHere"
          >
                    <i
                              className="material-icons md-18"
                    >
                              train
                    </i>
          </div>
</div>,
        <div
          className="stationCircle stationFilled"
>
           
</div>,
        <div
          className="stationInfo"
>
          <div
                    className=""
          >
                    Sevenoaks
          </div>
          <div
                    className="stationTimeInfo"
          >
                    On time
          </div>
</div>,
      ],
      "className": "stationContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "10:00",
            " ",
            <br />,
            <div
              className="stationTrainHere"
>
              <i
                            className="material-icons md-18"
              >
                            train
              </i>
</div>,
          ],
          "className": "stationTime",
        },
        "ref": null,
        "rendered": Array [
          "10:00",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons md-18"
>
                train
</i>,
              "className": "stationTrainHere",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "train",
                "className": "material-icons md-18",
              },
              "ref": null,
              "rendered": "train",
              "type": "i",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
          "className": "stationCircle stationFilled",
        },
        "ref": null,
        "rendered": " ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className=""
>
              Sevenoaks
</div>,
            <div
              className="stationTimeInfo"
>
              On time
</div>,
          ],
          "className": "stationInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sevenoaks",
              "className": "",
            },
            "ref": null,
            "rendered": "Sevenoaks",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "On time",
              "className": "stationTimeInfo",
            },
            "ref": null,
            "rendered": "On time",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stationTime"
>
            10:00
             
            <br />
            <div
                        className="stationTrainHere"
            >
                        <i
                                    className="material-icons md-18"
                        >
                                    train
                        </i>
            </div>
</div>,
          <div
            className="stationCircle stationFilled"
>
             
</div>,
          <div
            className="stationInfo"
>
            <div
                        className=""
            >
                        Sevenoaks
            </div>
            <div
                        className="stationTimeInfo"
            >
                        On time
            </div>
</div>,
        ],
        "className": "stationContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "10:00",
              " ",
              <br />,
              <div
                className="stationTrainHere"
>
                <i
                                className="material-icons md-18"
                >
                                train
                </i>
</div>,
            ],
            "className": "stationTime",
          },
          "ref": null,
          "rendered": Array [
            "10:00",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="material-icons md-18"
>
                  train
</i>,
                "className": "stationTrainHere",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "train",
                  "className": "material-icons md-18",
                },
                "ref": null,
                "rendered": "train",
                "type": "i",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
            "className": "stationCircle stationFilled",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className=""
>
                Sevenoaks
</div>,
              <div
                className="stationTimeInfo"
>
                On time
</div>,
            ],
            "className": "stationInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sevenoaks",
                "className": "",
              },
              "ref": null,
              "rendered": "Sevenoaks",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "On time",
                "className": "stationTimeInfo",
              },
              "ref": null,
              "rendered": "On time",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
